solana/indexer/checkpoints.go

```diff
@@ -0,0 +1,109 @@
+package indexer
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"bridgerton.audius.co/database"
+	"github.com/jackc/pgx/v5"
+	pb "github.com/rpcpool/yellowstone-grpc/examples/golang/proto"
+)
+
+func insertBackfillCheckpoint(ctx context.Context, db database.DBTX, fromSlot uint64, toSlot uint64, address string) (string, error) {
+	obj := map[string]string{
+		"type":    "backfill",
+		"address": address,
+	}
+	subscriptionJson, err := json.Marshal(obj)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal backfill subscription: %w", err)
+	}
+
+	sum := sha256.Sum256(subscriptionJson)
+	subscriptionHash := hex.EncodeToString(sum[:])
+
+	var checkpointId string
+	err = db.QueryRow(ctx, `
+			INSERT INTO sol_slot_checkpoints (from_slot, to_slot, subscription, subscription_hash) 
```

Comment:
Is it one row per subscription and the `to_slot` value gets updated as we process new incoming transactions? Or do we have a set of rows per subscription indicating chunks of transactions we processed?